modules/useQueries.js

```diff
@@ -26,25 +28,40 @@ function useQueries(createHistory) {
       parseQueryString = defaultParseQueryString
 
     function addQuery(location) {
-      if (location.query == null)
-        location.query = parseQueryString(location.search.substring(1))
+      if (location.query == null) {
+        const { search } = location
+        location.query = parseQueryString(search.substring(1))
+        location[SEARCH_BASE_KEY] = { search, searchBase: '' }
```

Comment:
I've read the code through a few times, but I'm still not sure what `searchBase` is or why it's necessary. Is there any way you can make this code more clear?